.github/workflows/build.yaml

```diff
@@ -46,22 +46,32 @@ jobs:
         with:
           name: metal-build-${{ matrix.os }}-${{ matrix.arch }}-${{ matrix.config.type }}
           path: ttm_${{ matrix.arch }}-${{ matrix.config.type}}.tar
-  build-cpptest:
+  build-other:
     strategy:
       matrix:
         config: [debug, release, assert]
         arch: [grayskull, wormhole_b0]
         os: [ubuntu-20.04]
+        test-group: [
+          {name: C++ tests, cmd: make tests},
+          {name: docs, cmd: ./tests/scripts/build_docs.sh},
```

Comment:
Why build docs on GH runners?

We build and run docs on post commit unit tests and on the separate docs build: https://github.com/tenstorrent-metal/tt-metal/actions/workflows/docs-latest-public.yaml

can we:

- get rid of the docs build in this workflow
- update the regular docs build to be post commit
- update the regular docs build to use artifacts (may have to play around with rpaths / LD_LIBRARY_PATH etc)